### ISSUE TITLE
False positive in identical_operands rule

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#2564]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

```
$ cat file.swift
let num: Int? = 0
_ = num != nil && num == num?.byteSwapped
$ cat .swiftlint.yml
included:
  - file.swift
whitelist_rules:
  - identical_operands
$ swiftlint version
0.29.4
$ swiftlint
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths
Linting 'file.swift' (1/1)
/.../file.swift:2:19: warning: Identical Operands Violation: Comparing two identical operands is likely a mistake. (identical_operands)
Done linting! Found 1 violation, 0 serious in 1 file.
```
